File: API/PageAndRDL/ReportWizard/VB.NET/ReportWizard/UI/WizardSteps/SelectSummaryOptions.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports GrapeCity.ActiveReports.Samples.ReportWizard.MetaData
Namespace UI.WizardSteps
	Partial Public Class SelectSummaryOptions
		Inherits BaseStep
		Public Sub New()
			InitializeComponent()
		End Sub
		Protected Overloads Overrides Sub OnDisplay(ByVal firstDisplay As Boolean)
			If firstDisplay Then
				selectedGroups.DisplayMember = "Title"
				outputFields.DisplayMember = "Title"

			End If
			masterReport.Text = ReportWizardState.SelectedMasterReport.Title
			Dim groups As New List(Of FieldMetaData)(ReportWizardState.GroupingFields)
			If ReportWizardState.GroupingFields IsNot Nothing Then
				If Not (ReportWizardState.DetailGroupingField) Is Nothing Then
					groups.Add(ReportWizardState.DetailGroupingField)
				End If
				Dim groupingFieldsbs As New BindingSource()
				groupingFieldsbs.DataSource = groups
				selectedGroups.DataSource = groupingFieldsbs
			End If
			If ReportWizardState.DisplayFields IsNot Nothing Then
				Dim displayFieldsbs As New BindingSource()
				displayFieldsbs.DataSource = ReportWizardState.DisplayFields
				outputFields.DataSource = displayFieldsbs
			End If
		End Sub
		Public Overloads Overrides Sub UpdateState()
			ReportWizardState.DisplayGrandTotals = grandTotal.Checked
			ReportWizardState.DisplayGroupSubtotals = subtotals.Checked
		End Sub
	End Class
End Namespace

File: API/PageAndRDL/ReportWizard/C#/ReportWizard/UI/WizardSteps/SelectSummaryOptions.cs
using System.Collections.Generic;
using System.Windows.Forms;
using GrapeCity.ActiveReports.Samples.ReportWizard.MetaData;

namespace GrapeCity.ActiveReports.Samples.ReportWizard.UI.WizardSteps
{
	public partial class SelectSummaryOptions : GrapeCity.ActiveReports.Samples.ReportWizard.UI.WizardSteps.BaseStep
	{
		public SelectSummaryOptions()
		{
			InitializeComponent();
		}

		protected override void OnDisplay( bool firstDisplay )
		{
			if (firstDisplay)
			{
				selectedGroups.DisplayMember = "Title";
				outputFields.DisplayMember = "Title";
			}
			
			masterReport.Text = ReportWizardState.SelectedMasterReport.Title;
			List<FieldMetaData> groups = new List<FieldMetaData>(ReportWizardState.GroupingFields);
			if (ReportWizardState.GroupingFields != null)
			{
				if (ReportWizardState.DetailGroupingField != null)
				{
					groups.Add(ReportWizardState.DetailGroupingField);
				}
				BindingSource groupingfieldsbs = new BindingSource();
				groupingfieldsbs.DataSource = groups;
				selectedGroups.DataSource = groupingfieldsbs;
				selectedGroups.DisplayMember = "Title";
			}
			if (ReportWizardState.DisplayFields != null)
			{
				BindingSource outputfieldsbs = new BindingSource();
				outputfieldsbs.DataSource = ReportWizardState.DisplayFields;
				outputFields.DataSource = outputfieldsbs;
				outputFields.DisplayMember = "Title";
			}
		}

		public override void UpdateState()
		{
			ReportWizardState.DisplayGrandTotals = grandTotal.Checked;
			ReportWizardState.DisplayGroupSubtotals = subtotals.Checked;
		}
	}
}
